File: ABC/080/b.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main(string[] args) {
        int n = int.Parse(Console.ReadLine());
        Console.WriteLine(n % digitSum(n) == 0 ? "Yes" : "No");
    }
    static int digitSum(int n) {
        Stack<int> digit = new Stack<int>();
        while(n > 0) {
            digit.Push(n%10);
            n /= 10;
        }
        return digit.ToArray().Sum();
    }
}

File: ABC/080/b.cpp
#include <bits/stdc++.h>
using namespace std;

int digitSum(int n) {
    int sum = 0;
    while(n>0){
        sum += n%10;
        n /= 10;
    }
    return sum;
}
int main() {
    int n; cin >> n;
    cout << (n%digitSum(n) == 0 ? "Yes" : "No") << endl;
}